src/bindings/text-buffer-wrapper.cc

```diff
@@ -527,14 +517,33 @@ void TextBufferWrapper::find_words_with_subsequence_in_range(const Nan::Function
 
     void HandleOKCallback() {
       delete snapshot;
-      Local<Array> js_result = Nan::New<Array>();
+      Local<Array> js_matches_array = Nan::New<Array>();
 
+      uint32_t positions_buffer_size = 0;
+      for (const auto &subsequence_match : result) {
+        positions_buffer_size += sizeof(uint32_t) + subsequence_match.positions.size() * sizeof(Point);
+      }
+
+      auto positions_buffer = v8::ArrayBuffer::New(v8::Isolate::GetCurrent(), positions_buffer_size);
+      uint32_t *positions_data = reinterpret_cast<uint32_t *>(positions_buffer->GetContents().Data());
+
+      uint32_t positions_array_index = 0;
       for (size_t i = 0; i < result.size() && i < max_count; i++) {
-        js_result->Set(i, SubsequenceMatchWrapper::from_subsequence_match(result[i]));
+        const SubsequenceMatch &match = result[i];
+        positions_data[positions_array_index++] = match.positions.size();
+        uint32_t bytes_to_copy = match.positions.size() * sizeof(Point);
+        memcpy(
+          reinterpret_cast<char *>(positions_data + positions_array_index),
```

Comment:
@maxbrunsfeld can you remind me why we casted to a `char` pointer here?

Reply:
Oh, perhaps that's not necessary! I thought `memcpy` needed a `char *` pointer, but it actually takes a `void *`, to which we can implicitly convert our pointer expression.

Reply:
Yea, it compiles and passes all the tests without it. I think I'll just remove it.